alephium/contracts/token_bridge/token_bridge_for_chain.ral

```diff
@@ -0,0 +1,69 @@
+Contract TokenBridgeForChain(
+    governanceContractId: ByteVec,
+    localChainId: U256,
+    localTokenBridgeId: ByteVec, // local chain token bridge contract id
+    remoteChainId: U256,
+    remoteTokenBridgeId: ByteVec, // remote chain token bridge contract id
+    mut next: U256,
+    mut next1: U256,
+    mut next2: U256,
+    undoneSequenceTemplateId: ByteVec, // used for create contract by copyCreateContract
+    refundAddress: Address,
+    mut sendSequence: U256
+) extends Sequence(next, next1, next2, undoneSequenceTemplateId, refundAddress), Constants(), TokenBridgeConstants() {
+    pub fn nextSendSequence() -> U256 {
+        checkPermission!(callerContractId!() == localTokenBridgeId, ErrorCodes.InvalidCaller)
+        let msgSequence = sendSequence
+        sendSequence = sendSequence + 1
+        return msgSequence
+    }
+
+    fn checkTransfer(vaa: ByteVec) -> (Bool, ByteVec) {
+        let governance = Governance(governanceContractId)
+        let (emitterChainId, targetChainId, emitterAddress, msgSequence, payload) = governance.parseAndVerifyVAA(vaa)
+        assert!(emitterChainId == remoteChainId, ErrorCodes.InvalidEmitChainId)
+        assert!(emitterAddress == remoteTokenBridgeId, ErrorCodes.InvalidEmitAddress)
+        assert!(targetChainId == localChainId, ErrorCodes.InvalidTargetChainId)
+        return checkSequence(msgSequence), payload
+    }
+
+    @using(readonly = true)
+    fn parseCompleteTransfer(payload: ByteVec) -> (U256, ByteVec, U256, Address, U256) {
+        assert!(byteVecSlice!(payload, 0, 1) == PayloadId.Transfer, ErrorCodes.InvalidPayloadId)
+        let amount = u256From32Byte!(byteVecSlice!(payload, 1, 33))
+        let vaaTokenId = byteVecSlice!(payload, 33, 65)
+        let vaaTokenChainId = u256From2Byte!(byteVecSlice!(payload, 65, 67))
+        let toAddress = byteVecSlice!(payload, 67, 99)
```

Comment:
This is because we changed the wormhole protocol. In the wormhole protocol, the messages sent by the token bridge of each chain to other chains will share the same sequence, which makes it difficult for us to deal with the sequence(because we use sequence instead of message hash to avoid replay attack). In the wormhole protocol, `toChainId` usually appears in the payload, we put `toChainId` in the header and changed ETH contracts in [this PR](https://github.com/alephium/wormhole-fork/pull/10), but our documentation has not been updated, there is [an issue](https://github.com/alephium/wormhole-fork/issues/12) to track this.

Reply:
I see, thanks for the great explanation!